### PR TITLE
Container generation via distrobuilder

### PR DESCRIPTION
**Pull request by HexF**
Distrobuilder this time - Automatically builds LXC containers and metadata.

Carries on from #60

### CODE REVIEW

**Comment by HexF** [comment]
Should be ready for merge as soon as distrobuilder finishes its thing - it takes quite a while

File: container/piston.yaml [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+    [ "$(uname -m)" = "aarch64" ] && TARGET="arm64"
+    update-grub
+    grub-install --uefi-secure-boot --target="${TARGET}-efi" --no-nvram --removable
+    update-grub
+    sed -i "s#root=[^ ]*#root=/dev/sda2#g" /boot/grub/grub.cfg
+  types:
+  - vm
+
+mappings:
+  architecture_map: debian
```
Comment (TroyDota) [inline container/piston.yaml:355]: Add a new line?

File: container/configure.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+with open('piston.yaml') as dbc:
+    with open('install_script.sh') as install_script_file:
+        with open('build.yaml' , 'w+') as distrobuilder_config_file_new:
+            distrobuilder_config = yaml.safe_load(dbc)
+            distrobuilder_config['actions'].append({
+                'trigger': 'post-packages',
+                'action': install_script_file.read(),
+
+            })
+            yaml.dump(distrobuilder_config, distrobuilder_config_file_new)
```
Comment (TroyDota) [inline container/configure.py:13]: Add a new line?

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,5 @@
 api/api
 lxc/i
 lxc/lockfile
+container/build.yaml
+container/*.tar.xz
```
Comment (TroyDota) [inline .gitignore:5]: Add a new line?

File: .github/workflows/lxc-build.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+          sudo chown -R runner:runner .
+        working-directory: container
+      
+      #  Upload container 
+      - uses: actions/upload-artifact@v2
+        with:
+          name: lxc-container
+          path: |
+            container/output-piston-bionic/meta.tar.xz
+            container/output-piston-bionic/rootfs.tar.xz
```
Comment (TroyDota) [inline .github/workflows/lxc-build.yml:47]: Add a new line?

**Comment by HexF** [comment]
Automated builds are proving a challenge, Maybe its best if we let people just run the `build.sh` scripts themselves until we can get CI working. 
Travis CI won't work because they only upload artifacts externally from what I have read, and GitHub Actions fills the disk, causing it to error out

**Comment by HexF** [comment]
Should be ready for a merge now

**Pull request merged**